src/app/pages/rpc-stx-sign-transaction/rpc-stx-sign-transaction.tsx

```diff
@@ -1,29 +1,156 @@
+import { ContractDeployDetailsLayout } from '@app/features/rpc-transaction-request/stacks/contract-deploy/contract-deploy-details.layout';
+import { PostConditionsDetailsLayout } from '@app/features/rpc-transaction-request/stacks/post-conditions/post-conditions-details.layout';
+import { useStacksRpcTransactionRequestContext } from '@app/features/rpc-transaction-request/stacks/stacks-rpc-transaction-request.context';
+import { SwitchAccountTrigger } from '@app/features/rpc-transaction-request/switch-account-trigger/switch-account-trigger';
+import { TransactionActionsWithSpend } from '@app/features/rpc-transaction-request/transaction-actions/transaction-actions-with-spend';
+import { useSignStacksTransaction } from '@app/store/transactions/transaction.hooks';
+
+import {
+  checkUnsignedStacksTransactionFee,
+  checkUnsignedStacksTransactionHashMode,
+  checkUnsignedStacksTransactionNonce,
+  getUnsignedStacksTransactionFromRpcRequest,
+} from './rpc-stx-sign-transaction.utils';
 
 export function RpcStxSignTransaction() {
-  const {
-    onSignStacksTransaction,
-    disableFeeSelection,
-    stacksTransaction,
-    disableNonceSelection,
-    isMultisig,
-    txSender,
-  } = useRpcStxSignTransaction();
+  const { address, isLoadingBalance, onUserActivatesSwitchAccount, requestId, tabId } =
+    useStacksRpcTransactionRequestContext();
+  const { availableBalance, isLoadingFees, marketData, onUserActivatesFeeEditor, selectedFee } =
+    useFeeEditorContext();
+  const { nonce, onUserActivatesNonceEditor } = useNonceEditorContext();
+  const convertToFiatAmount = useConvertCryptoCurrencyToFiatAmount('STX');
+  const signStacksTx = useSignStacksTransaction();
+
+  const unsignedTxForBroadcast = useMemo(() => getUnsignedStacksTransactionFromRpcRequest(), []);
+  const txRequestHasAlreadySetFee = checkUnsignedStacksTransactionFee(unsignedTxForBroadcast);
+  const txRequestHasAlreadySetNonce = checkUnsignedStacksTransactionNonce(unsignedTxForBroadcast);
+
+  // Handle multisig transactions
+  const isMultisig = checkUnsignedStacksTransactionHashMode(unsignedTxForBroadcast);
+  const txRequestWasAlreadySignedByOthers =
+    isMultisig &&
+    'fields' in unsignedTxForBroadcast.auth.spendingCondition &&
+    unsignedTxForBroadcast.auth.spendingCondition.fields.length > 0;
+  const enableFeeEditor = !(txRequestWasAlreadySignedByOthers || txRequestHasAlreadySetFee);
+  const enableNonceEditor = !(txRequestWasAlreadySignedByOthers || txRequestHasAlreadySetNonce);
+
+  async function onApproveTransaction() {
+    if (!txRequestHasAlreadySetFee)
+      unsignedTxForBroadcast.setFee(selectedFee.txFee.amount.toString());
+    if (!txRequestHasAlreadySetNonce) unsignedTxForBroadcast.setNonce(nonce);
+
+    const signedTransaction = await signStacksTx(unsignedTxForBroadcast);
+    if (!signedTransaction) throw new Error('Error signing stacks transaction');
```

Comment:
Shouldn't we return something to the app in this case?

Reply:
I don't see that we've ever returned anything to the app in this case, should we ...and should I add an `RpcErrorCode` for it, or handle separately? 🤔

Reply:
I think we can use `RpcErrorCode.INVALID_REQUEST` here with a message that the tx wasn't signed.